### PR TITLE
Updating project for strict concurrency and Swift 6

### PR DESCRIPTION
**Pull request by kylebrowning**

### CODE REVIEW

File: Sources/APNSCore/APNSClient.swift [hunk 1]
```diff
@@ -12,6 +12,9 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Dispatch
+
 public protocol APNSClientProtocol {
     func send(_ request: APNSRequest<some APNSMessage>) async throws -> APNSResponse
+    func shutdown(queue: DispatchQueue, callback: @Sendable @escaping (Error?) -> Void)
```
Comment (0xTim) [inline Sources/APNSCore/APNSClient.swift:19]: Why not make this async?

File: Sources/APNSExample/Program.swift [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 import APNSCore
 import APNS
+import OSLog
```
Comment (0xTim) [inline Sources/APNSExample/Program.swift:17]: Should we use SwiftLog here instead?

File: Package.swift [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.7
+// swift-tools-version:5.9
```
Comment (0xTim) [inline Package.swift:1]: Assuming this is going into the major release why not 5.10 (or just 6.0 directly)

File: Sources/APNS/APNSClient.swift [hunk 4]
```diff
@@ -125,14 +125,9 @@ public final class APNSClient<Decoder: APNSJSONDecoder, Encoder: APNSJSONEncoder
     /// - Parameters:
     ///   - queue: The queue on which the callback is invoked on.
     ///   - callback: The callback that is invoked when everything is shutdown.
-    @preconcurrency public func shutdown(queue: DispatchQueue = .global(), callback: @Sendable @escaping (Error?) -> Void) {
+    public func shutdown(queue: DispatchQueue = .global(), callback: @Sendable @escaping (Error?) -> Void) {
```
Comment (0xTim) [inline Sources/APNS/APNSClient.swift:128]: See comment below, you could change this to

```swift
try await self.httpClient.shutdown()
```

This makes all the dispatch stuff an AHC problem

Comment (kylebrowning) [reply Sources/APNS/APNSClient.swift:128]: Got it, done

File: Package.swift (hunk 3 above)
Comment (kylebrowning) [reply Package.swift:1]: Yeah, sure

File: Sources/APNSExample/Program.swift (hunk 2 above)
Comment (kylebrowning) [reply Sources/APNSExample/Program.swift:17]: Dusting off my chops sorry haah

File: Sources/APNSCore/APNSClient.swift (hunk 1 above)
Comment (kylebrowning) [reply Sources/APNSCore/APNSClient.swift:19]: DOne

File: Package.swift [hunk 5]
```diff
@@ -31,17 +31,26 @@ let package = Package(
...
+        ),
         .testTarget(
             name: "APNSTests",
             dependencies: [
                 .target(name: "APNSCore"),
                 .target(name: "APNS"),
-            ]),
+            ],
+            swiftSettings: [
+              .enableExperimentalFeature("StrictConcurrency")
```
Comment (0xTim) [inline Package.swift:44]: If you're going for a 6.0 manifest, then it's no longer an experimental feature. Best thing to do is set the language mode for the package like https://github.com/vapor/jwt-kit/blob/main/Package.swift#L43

File: Sources/APNS/APNSClient.swift [hunk 6]
```diff
@@ -114,24 +114,9 @@ public final class APNSClient<Decoder: APNSJSONDecoder, Encoder: APNSJSONEncoder
...
-    @preconcurrency public func shutdown(queue: DispatchQueue = .global(), callback: @Sendable @escaping (Error?) -> Void) {
-        self.httpClient.shutdown(callback)
-    }
-
-    /// Shuts down the client and `EventLoopGroup` if it was created by the client.
-    public func syncShutdown() throws {
-        try self.httpClient.syncShutdown()
+    /// Shuts down the client and event loop gracefully.
+    public func shutdown() async throws {
+        try await self.httpClient.shutdown()
```
Comment (0xTim) [inline Sources/APNS/APNSClient.swift:119]: Does this compile? It might need a `.get()` at the end

File: Sources/APNSCore/APNSClient.swift [hunk 7]
```diff
@@ -12,6 +12,9 @@
 //
 //===----------------------------------------------------------------------===//
 
+import Dispatch
```
Comment (0xTim) [inline Sources/APNSCore/APNSClient.swift:15]: No longer needed

**Review by 0xTim — CHANGES_REQUESTED** [review]

File: Sources/APNS/APNSClient.swift (hunk 6 above)
Comment (kylebrowning) [reply Sources/APNS/APNSClient.swift:119]: It does compile

File: Package.swift (hunk 5 above)
Comment (kylebrowning) [reply Package.swift:44]: thx, done

File: Sources/APNS/APNSClient.swift [hunk 8]
```diff
@@ -114,24 +113,9 @@ public final class APNSClient<Decoder: APNSJSONDecoder, Encoder: APNSJSONEncoder
...
-    ///   - queue: The queue on which the callback is invoked on.
-    ///   - callback: The callback that is invoked when everything is shutdown.
-    @preconcurrency public func shutdown(queue: DispatchQueue = .global(), callback: @Sendable @escaping (Error?) -> Void) {
-        self.httpClient.shutdown(callback)
-    }
-
-    /// Shuts down the client and `EventLoopGroup` if it was created by the client.
-    public func syncShutdown() throws {
-        try self.httpClient.syncShutdown()
+    /// Shuts down the client and event loop gracefully.
```
Comment (0xTim) [inline Sources/APNS/APNSClient.swift:116]: Nit - this doesn't touch the event loop

File: Sources/APNSExample/Program.swift [hunk 9]
```diff
@@ -14,8 +14,11 @@
 
 import APNSCore
 import APNS
+import Logging
 import Foundation
 
+let logger = Logger(label: "APNSwiftExample")
```
Comment (0xTim) [inline Sources/APNSExample/Program.swift:20]: You have this logger and another one declared below

**Review by 0xTim — CHANGES_REQUESTED** [review]
Couple of minor points

File: Sources/APNSExample/Program.swift (hunk 9 above)
Comment (kylebrowning) [reply Sources/APNSExample/Program.swift:20]: Fixed! Thank

**Review by 0xTim — APPROVED** [review]
LGTM!

**Pull request merged**